### PR TITLE
chore: App router migration - Fix `/event-types/[type]` page, move `getServerSideProps` to `/lib` for event-types pages

### PR DESCRIPTION
**Pull request by hbjORbj**
## What does this PR do?

- Fixes CAL-4247, CAL-4308
- Removes references to `/pages` for `/event-types`
-  Fix `/event-types/[type]` page by creating a copy of `EventType` where outdated methods of router from `next/router` are replaced with `usePathname`

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Navigate to:

1. `future/event-types/[type]`
2. `future/event-types`
3. `/event-types/[type]`
4. `/event-types`

### CODE REVIEW

**Review by Udit-takkar — DISMISSED** [review]
I am getting these errors on /future/event-types/[type] "NextRouter was not mounted."

Screenshot 2024-09-07 at 12 46 26 AM

**Comment by hbjORbj** [comment]
Hey Udit, I couldn't reproduce the error you had. BUT thanks to you, I found one thing that was bothering; we were importing useRouter from next/router, which we shouldn't for app router. So, this means, I need to create a copy of this `EventType` component and refactor it for app router. I will convert it to a draft and ping you once it's done ;)

**Comment by hbjORbj** [comment]
@Udit-takkar Hey Udit, thanks for the review. The problem was that router was being created from `useRouter` imported from `next/router`, not the one from `next/navigation`. I created a copy of `EventType` component called `EventTypeAppDir`. Now it should render fine.

Screenshot 2024-09-10 at 22 04 47

File: packages/features/eventtypes/components/EventTypeAppDir.tsx [hunk 1]
```diff
@@ -0,0 +1,755 @@
...
+        setPendingRoute(url);
+        // The part where it is different from the original code in `EventType` component STARTS HERE
+        throw new Error(`Aborted route change to ${url} because none was assigned to team event`);
+      }
+    };
+    handleRouteChange(pathname);
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [pathname, eventType.hosts, eventType.children, eventType.assignAllTeamMembers]);
+  // The part where it is different from the original code in `EventType` component ENDS HERE
```
Comment (hbjORbj) [inline packages/features/eventtypes/components/EventTypeAppDir.tsx:335]: FYI

**Review by zomars — DISMISSED** [review]

**Review by zomars — APPROVED** [review]

**Pull request merged**